lib/src/models/address_params_enterprise.dart

```diff
@@ -0,0 +1,15 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:ledger_cardano/src/models/spending_params.dart';
+
+part 'address_params_enterprise.freezed.dart';
+
+@freezed
+sealed class AddressParamsEnterprise with _$AddressParamsEnterprise {
+   AddressParamsEnterprise._();
+  
+
+  factory AddressParamsEnterprise({
+    required SpendingParams spendingPath,
+  }) = _AddressParamsEnterprise;
+}
```

Comment:
If you don't have extra `fields`/`methods` added to freezed class, the constructor and the factory can both be `const`